frappe/database/schema.py

```diff
@@ -208,6 +221,9 @@ def get_definition(self, for_modification=False):
 		):
 			column_def += f" default {frappe.db.escape(self.default)}"
 
+		elif self.not_nullable:
+			column_def += f" not null default {frappe.db.escape('')}"
```

Comment:
Umm, default is already set on line above.

Reply:
Hmm, I guess I should split out this from the above if-elif chain and keep it standalone like `unique`.